### DISCUSSION TITLE
Terminal connection crashes in deepsleep

### DISCUSSION THREAD

**Discussion by KrTamir** [micropython/micropython#15323] [General]
Hello all,
i was using a various types of ESP32. (for exmple S2 mini)
enviroment: VScode , windows , frimwork - LOLIN_C3_MINI-20240222-v1.22.2.bin

while using ampy in the terminal of VScode
whenever i used the deepsleep option the connection broke and no more prints were made.
the board continue to do the script but i can not debug any more.

right now i get:
"Failed to find or read input file: C:\Users\..."
in the past i got somthing of EOF error

any idieas how to solve it \ debug in a smrter\better way?

**Comment by davefes** [comment]
Unless there has been any previous suggestions, that I have missed ... do all your debugging without deep or lightsleep and then when you are happy enable them.

Also, if you want to print  messages before going to deepsleep (maybe lightsleep as well) put this after your print statement:
`time.sleep_ms(10)`

**Comment by kjm1102** [comment]
The new crop of esp32 descendants with USB baked in are not as well behaved with light/deep  sleep in repl as their predecessors that used an external uart-to-USB chip. The problem is further complicated by the fact that the esp32 runs user code from scratch after deepsleep rather than from where the program pointer was when it went into deepsleep. So when you're substituting sleep for deepsleep during debugging you have to be mindful that program flow will be altered when you replace sleep with deepsleep. In this respect the rpi2040 is easier to work with since program resumption after sleep or deeepsleep is the same.

**Reply by robert-hh** [reply]
The rp2040 code was changed to start from reset after deepsleep.

**Reply by kjm1102** [reply]
Ah OK, what version was that change? I kinda liked the way the rp2040 just picked up from where it left off with all the metrics in ram still intact. Struck me as more logical than the esp32 reboot after deepsleep.

**Reply by robert-hh** [reply]
> Ah OK, what version was that change? 

The respective commit is from February 2nd, 2021.
